location/index.js

```diff
@@ -76,29 +76,39 @@ class IRNMNLocation extends HTMLElement {
         return this.getAttribute('error-message') || 'Please select a valid location';
     }
 
+
     render() {
+        if (!this.locations) {
+            console.error("No locations provided");
+            return;
+        }
+    
         this.innerHTML = `
             <div class="${CLASS_NAMES.container}">
                 <label for="${this.inputId}" class="${CLASS_NAMES.label}">${this.label}</label>
                 <select id="${this.inputId}" name="${this.inputName}" class="${CLASS_NAMES.select}">
                     <option value="" disabled selected>${this.placeholder}</option>
-                    ${this.locations.map(location => 
-                        `<option value="${location.hotelCode}" 
-                            data-max-rooms="${location.maxRooms}" 
-                            data-min-rooms="${location.minRooms}"
-                            data-open-date="${location.openDate}"
-                            data-date-display-format="${location.dateDisplayFormat}"
-                            data-date-locale="${location.dateLocale}"
-                            class="${CLASS_NAMES.option}">
-                            ${location.name}
-                        </option>`
-                    ).join('')}
```

Comment:
The data attributes are now populated based on the location object. 
advantage: If we create a new component with new parameters, we don't need to do any changes in the location one.